restapi.module

```diff
@@ -374,10 +377,13 @@ function restapi_delivery_callback($response) {
 
   // For requests in the browser, we'll try to show the output in a more
   // pleasant way.
-  if (!$request->isJson() &&
-      strpos($response->getHeaderLine('content-type'), 'application/json') !== FALSE &&
-      strpos($request->getHeaderLine('Accept'), 'text/html') !== FALSE) {
-    return restapi_deliver_html($response);
+  $negotiator = new Negotiator();
+  $accept     = $request->getHeaderLine('Accept');
+  $priorities = ['application/json', 'text/html'];
+
+  if ($negotiator->getBest($accept, $priorities) === 'text/html') {
```

Comment:
The functional change here is if the user agent accepts both JSON and HTML, but *prefers* JSON over HTML, restapi will not deliver the pretty HTML version of the response.